src/middlewared/middlewared/plugins/system.py

```diff
@@ -152,6 +155,328 @@ def debug(self, job):
+                    )
+                    if len(certificate_obj) == 0:
+                        verrors.add(
+                            f'{schema}.guicertificate',
+                            'No matching certificate found in database records, kindly check again'
+                        )
+                        raise verrors
+
+                    certificate_obj = certificate_obj[0]
+                    # getting fingerprint for certificate
+                    fingerprint = await self.middleware.call(
+                        'certificate.get_certificate_fingerprint',
+                        certificate_obj['cert_certificate']
+                    )
+                    if fingerprint:
+                        syslog.openlog(logoption=syslog.LOG_PID, facility=syslog.LOG_USER)
+                        syslog.syslog(syslog.LOG_NOTICE, 'Fingerprint of the certificate used in UI : ' + fingerprint)
+                        syslog.closelog()
+        return verrors
+
+    @accepts(
+        Dict(
+            'general_settings',
+            IPAddr('uiaddress'),
+            Int('uicertificate'),
+            Int('uihttpsport', validators=[Range(min=1, max=65535)]),
+            Bool('uihttpsredirect'),
+            Int('uiport', validators=[Range(min=1, max=65535)]),
+            Str('uiprotocol', enum=['HTTP', 'HTTPS', 'HTTPHTTPS']),
+            IPAddr('uiv6address'),
+            Str('kbdmap'),
+            Str('language'),
+            Str('sysloglevel', enum=['F_EMERG', 'F_ALERT', 'F_CRIT', 'F_ERR', 'F_WARNING', 'F_NOTICE',
+                                     'F_INFO', 'F_DEBUG', 'F_IS_DEBUG']),
+            Str('syslogserver', required=True),
+            Str('timezone')
+        )
+    )
+    async def do_update(self, data):
+        data['sysloglevel'] = data['sysloglevel'].lower()
+        data['uiprotocol'] = data['uiprotocol'].lower()
+
+        config = await self.config()
+        new_config = config.copy()
+        new_config.update(data)
+        verrors = await self.validate_general_settings(new_config, 'general_settings_update')
+        if verrors:
+            raise verrors
+
+        keys = new_config.keys()
+        for key in keys:
+            if isinstance(key, str) and key.startswith('ui'):
+                new_config['g' + key] = new_config.pop(key)
+
+        await self.middleware.call('datastore.update', 'system.settings', config['id'], new_config, {'prefix': 'stg_'})
+
+        if (
+            config['sysloglevel'] != new_config['sysloglevel'] or
+                config['syslogserver'] != new_config['syslogserver']
+        ):
+            await self.middleware.call('service.restart', 'syslogd')
+
+        await self.middleware.call('service.reload', 'timeservices')
+
+        if config['timezone'] != new_config['timezone']:
+            await self.middleware.call('service.restart', 'cron')
+
+        await self.middleware.call('service._start_ssl', 'nginx')
+        return new_config
```

Comment:
This will return the new config as it is in the database, perhaps just use self.config() again?